core/meterer/offchain_store.go

```diff
@@ -199,14 +180,15 @@ func (s *OffchainStore) GetRelevantOnDemandRecords(ctx context.Context, accountI
 	}
 	smallerResult, err := s.dynamoClient.QueryWithInput(ctx, queryInput)
 	if err != nil {
-		return 0, 0, 0, fmt.Errorf("failed to query smaller payments for account: %w", err)
+		return nil, nil, 0, fmt.Errorf("failed to query smaller payments for account: %w", err)
 	}
-	var prevPayment uint64
+	prevPayment := big.NewInt(0)
 	if len(smallerResult) > 0 {
-		prevPayment, err = strconv.ParseUint(smallerResult[0]["CumulativePayments"].(*types.AttributeValueMemberN).Value, 10, 64)
-		if err != nil {
-			return 0, 0, 0, fmt.Errorf("failed to parse previous payment: %w", err)
+		setPrevPayment, success := prevPayment.SetString(smallerResult[0]["CumulativePayments"].(*types.AttributeValueMemberN).Value, 10)
```

Comment:
I know these values are all expected in the happy path, but this chain seems dangerous
It's missing at least a couple validations: checking if `"CumulativePayments"` exists in the map and the result of the type assertion

Here and below as well

Reply:
added explicit checking for them!